src/main/scala/com/ossuminc/riddl/plugins/idea/ui/RiddlToolWindowFactory.scala

```diff
@@ -113,17 +130,24 @@ class RiddlToolWindowContent(
 
     val confFile = File(statePath)
 
-    val output = getRiddlIdeaState.getState.riddlOutput
+    val output = state.riddlOutput
     if output.nonEmpty then
       outputLabel.setText(
         s"<html>${output.mkString("<br>")}</html>"
       )
     else if fromReload || (confFile.exists() && confFile.isFile) then
-      parseASTFromConfFile(statePath)
+      parseASTFromConfFile(numWindow, statePath)
     else
       outputLabel.setText(
         s"<html>File: " + statePath +
           "<br>riddlc: project's .conf file not found, please configure in setting</html>"
       )
   }
+
+  def createToolWindow(): Unit = {
+    toolWindow.createAndAddContentToTW(
+      project,
+      getRiddlIdeaStates.newState()
+    )
+  }
```

Comment:
I have a design comment about this. The 2nd parameter is the state. You're trying to hold that in a "RiddlIdeaSettings" object a single class that holds all of the window states. I think this is problematic because it doesn't  associate the windows state with the plugin state one-to-one.   You should be able to just create a RiddlIdeaSettings.State class directly on line 150.  So, I think what is needed is "RiddlIdeaSettings.State" >class<, like you had previously,  that holds the state for one window. No need to have a list of all the states that you have to maintain. You should be able to ask the `toolWindow` for the state (second argument to createAndAddContentToTW)?